packages/edit-widgets/src/store/transformers.js

```diff
@@ -14,75 +14,68 @@ function __experimentalAddWidgetIdToBlock( block, widgetId ) {
 }
 
 export function transformWidgetToBlock( widget ) {
-	if ( widget.widget_class === 'WP_Widget_Block' ) {
-		const parsedBlocks = parse( widget.settings.content );
+	if ( widget.id_base === 'block' ) {
+		const parsedBlocks = parse( widget.instance.raw.content );
 		if ( ! parsedBlocks.length ) {
-			return __experimentalAddWidgetIdToBlock(
+			return addWidgetIdToBlock(
 				createBlock( 'core/paragraph', {}, [] ),
 				widget.id
 			);
 		}
-		return __experimentalAddWidgetIdToBlock( parsedBlocks[ 0 ], widget.id );
+		return addWidgetIdToBlock( parsedBlocks[ 0 ], widget.id );
 	}
 
-	const attributes = {
-		name: widget.name,
-		form: widget.form,
-		instance: widget.settings,
-		idBase: widget.id_base,
-		number: widget.number,
-	};
-
-	const isReferenceWidget = ! widget.widget_class;
-	if ( isReferenceWidget ) {
-		attributes.referenceWidgetName = widget.id;
+	let attributes;
+	// TODO: Don't use widget_class.
+	if ( widget.widget_class ) {
+		attributes = {
+			idBase: widget.id_base,
+			instance: widget.instance,
+		};
 	} else {
-		attributes.widgetClass = widget.widget_class;
+		attributes = {
+			id: widget.id,
+		};
 	}
 
-	return __experimentalAddWidgetIdToBlock(
+	return addWidgetIdToBlock(
 		createBlock( 'core/legacy-widget', attributes, [] ),
 		widget.id
 	);
 }
 
 export function transformBlockToWidget( block, relatedWidget = {} ) {
-	const { name } = block;
-	if ( name === 'core/legacy-widget' ) {
-		let widget;
-		if ( block.attributes.referenceWidgetName ) {
-			widget = transformReferenceBlockToWidget( block, relatedWidget );
-		} else {
-			widget = transformClassBlockToWidget( block, relatedWidget );
-		}
-		delete widget.form;
-		delete widget.rendered;
-		return widget;
+	let widget;
+
+	if ( block.name === 'core/legacy-widget' ) {
+		widget = {
+			...relatedWidget,
+			id: block.attributes.id ?? relatedWidget.id,
+			id_base: block.attributes.idBase ?? relatedWidget.id_base,
+			instance: block.attributes.instance ?? relatedWidget.instance,
+		};
+	} else {
+		widget = {
+			...relatedWidget,
+			id_base: 'block',
+			instance: {
+				raw: {
+					content: serialize( block ),
+				},
+			},
+		};
 	}
 
-	return {
-		...relatedWidget,
-		id_base: 'block',
-		widget_class: 'WP_Widget_Block',
-		settings: {
-			content: serialize( block ),
-		},
-	};
-}
+	// Delete deprecated properties.
+	delete widget.description;
+	delete widget.name;
+	delete widget.number;
+	delete widget.settings;
+	delete widget.widget_class;
 
-function transformClassBlockToWidget( { attributes }, relatedWidget ) {
-	return {
-		...relatedWidget,
-		widget_class: attributes.widgetClass,
-		id_base: attributes.idBase,
-		settings: attributes.instance,
-	};
-}
+	// Delete read-only properties.
+	delete widget.rendered;
+	delete widget.rendered_form;
```

Comment:
Instead of deleting deprecated and read-only properties, couldn't we just return a new object with the required properties?

Reply:
The benefit of doing it this way is that the REST API is in control of what properties the object has. If we return a new object then we'd have to add new properties to the object if/when the REST API adds a new property.

We can remove the `// Delete deprecated properties.` stuff in Gutenberg 10.5 which will clean this up.

Reply:
Okay, makes sense to me 👍